consumers/campaigner/main.go

```diff
@@ -44,20 +47,41 @@ func (h *MessageHandler) HandleMessage(m *nsq.Message) error {
 		return nil
 	}
 
-	// fetching subs that are active and that have not been blacklisted
+	campaign, err := h.s.GetCampaign(msg.CampaignID, msg.UserID)
+	if err != nil {
+		logrus.WithFields(logrus.Fields{
+			"campaign_id": msg.CampaignID,
+			"user_id":     msg.UserID},
+		).WithError(err).Error("unable to find campaign")
+		return nil
+	}
+	if campaign.Status != entities.StatusDraft {
+		logrus.WithError(err).
+			WithField("status", campaign.Status).
+			Warn("this is duplicate message in campaigner consumer")
+		return nil
+	}
+
+	campaign.Status = entities.StatusSending
+	err = h.s.UpdateCampaign(campaign)
+	if err != nil {
+		logrus.WithError(err).
+			WithField("campaign", campaign).
```

Comment:
Not sure how campaign will render here. It's better to just add separate fields like `user_id`, `campaign_id` instead of a whole object.